server/controllers/groups.js

```diff
@@ -552,70 +477,27 @@ module.exports = function(app) {
    * Get group content.
    */
   const getOne = (req, res, next) => {
-    async.auto({
-
-      getBalance: getBalance.bind(this, req.group.id),
-      getYearlyIncome: getYearlyIncome.bind(this, req.group.id),
-      getPublicPageInfo: getPublicPageInfo.bind(this, req.group.id),
-
-      getActivities: (cb) => {
-        if (!req.query.activities && !req.body.activities) {
-          return cb();
-        }
-
-        const query = {
-          where: {
-            GroupId: req.group.id
-          },
-          order: [['createdAt', 'DESC']],
-          offset: 0,
-          limit: 20 // [TODO] I need to put this default value
-          // as a global parameter. Using mw.paginate?
-        };
-
-        Activity
-          .findAndCountAll(query)
-          .then((activities) => cb(null, activities.rows))
-          .catch(cb);
-      },
-
-      getStripeAccount: (cb) =>  {
-        req.group.getStripeAccount()
-          .then((account) => cb(null, account))
-          .catch(cb);
-      },
-
-      getConnectedAccount: (cb) => {
-        req.group.getConnectedAccount()
-          .then((account) => cb(null, account))
-          .catch(cb);
-      }
-
-    }, (e, results) => {
-      if (e) return next(e);
-
-      const group = _.extend({}, req.group.info, {
-        balance: results.getBalance,
-        yearlyIncome: results.getYearlyIncome,
-        backersCount: results.getPublicPageInfo.backersCount,
-        donationTotal: results.getPublicPageInfo.donationTotal
-      });
-
-      if (results.getActivities) {
-        group.activities = results.getActivities;
-      }
-
-      if (results.getStripeAccount) {
-        group.stripeAccount = _.pick(results.getStripeAccount, 'stripePublishableKey');
-      }
-
-      if (_.get(results, 'getConnectedAccount.provider') === 'paypal') {
-        group.hasPaypal = true; // hack for the prototype, to refactor
-      }
-
-      res.send(group);
-    });
-
+    const group = req.group.info;
+
+    Promise.all([
+      req.group.getStripeAccount(),
+      req.group.getConnectedAccount(),
+      req.group.getBalance(),
+      req.group.getYearlyIncome(),
+      req.group.getTotalDonations(),
+      req.group.getBackersCount()
+      ])
+    .then(values => {
+      group.stripeAccount = values[0] && _.pick(values[0], 'stripePublishableKey');
+      group.hasPaypal = values[1] && values[1].provider === 'paypal';
+      group.balance = values[2];
+      group.yearlyIncome = values[3];
+      group.donationTotal = values[4];
+      group.backersCount = values[5];
+      return group;
+    })
+    .then(group => res.send(group))
```

Comment:
I tried that. For somer reason, it didn't work. Makes me wonder if there is something else wrong with this promise chain.